plugins/modules/lxd_container.py

```diff
@@ -551,6 +556,8 @@ def _create_instance(self):
             url = '{0}?{1}'.format(url, urlencode(url_params))
         config = self.config.copy()
         config['name'] = self.name
+        if 'instances' in url:
```

Comment:
I would set something like `self.uses_instances = True` in `__init__` and then do
```suggestion
        if self.uses_instances:
```
here. That's a lot cleaner than relying on some magic string in an URL, which could also appear in the hostname or any other part of the URL that is controlled by the user.

Reply:
When I first wrote this I was only attempting to match on the `self.api_endpoint` but due to how the url is being built up I can agree that `instances` could appear elsewhere in the URL. With the api endpoint only ever being `/1.0/container`, `/1.0/virtual-machines`, or `/1.0/instances` I think check if it contains `instances` isn't a bad approach. Now, moving it up into __init__ might be a good idea (I'm pretty new to Python so excuse my ignorance)

Reply:
`url` can have quite a few more values, since `url_params` has been added. So it's better to not look for `instances` in `url`.

Reply:
```suggestion
        if self.uses_instances:
```

Reply:
I added suggestions on how it could look like.